File: frontend/ediary_cli/src/main/kotlin/org/srd/ediary_cli/model/Diary.kt
package org.srd.ediary_cli.model

import kotlinx.datetime.LocalDate
import kotlinx.serialization.Serializable

@Serializable
data class DiaryCreateDTO(val ownerID: Long, val title: String, val description: String)

@Serializable
data class DiaryInfoDTO(
    val id: Long,
    val title: String,
    val description: String,
    val cntEntry: Int,
    val createdDate: LocalDate
)

File: frontend/ediary_vue/src/types/Diary.ts
export interface Diary {
    id: number;
    ownerID: number;
    title: string;
    description: string;
    cntEntries: number;
    createdDate: string;
}

export interface DiaryInfoDTO {
    id: number;
    title: string;
    description: string;
    cntEntries: number;
    createdDate: string;
}

export interface DiaryCreateDTO {
    ownerID: number;
    title: string;
    description: string;
}
